File: vita/src/main/java/vita/ep/m4/finalized/Mult.java
package vita.ep.m4.finalized;

import vita.ep.m4.Exp;

public class Mult extends Factory implements vita.ep.m4.Mult<Visitor> {

    protected Exp<Visitor> left;

    protected Exp<Visitor> right;

    public Mult(Exp<Visitor> _left, Exp<Visitor> _right) {
        this.left = _left;
        this.right = _right;
    }

    public Exp<Visitor> getLeft() {
        return this.left;
    }

    public Exp<Visitor> getRight() {
        return this.right;
    }

    public void accept(Visitor visitor) {
        visitor.visit(this);
    }

    public Exp<Visitor> convert(vita.ep.Exp<Visitor> other) {
        Visitor visitor = new Visitor();
        other.accept(visitor);
        return visitor.getResult();
    }
}

File: coco_csharp/CoCo/EP/M3/Finalized/Mult.cs
namespace EP.M3.Finalized {
    public class Mult : EP.M2.Finalized.Exp, EP.M3.Mult<EP.M2.Finalized.Exp>, Factory {
        private EP.Exp<EP.M2.Finalized.Exp> left;
        private EP.Exp<EP.M2.Finalized.Exp> right;

        public Mult(EP.Exp<EP.M2.Finalized.Exp> _left, EP.Exp<EP.M2.Finalized.Exp> _right) {
            this.left = _left;
            this.right = _right;
        }

        public EP.Exp<EP.M2.Finalized.Exp> getLeft() { return this.left; }

        public EP.Exp<EP.M2.Finalized.Exp> getRight() { return this.right; }
    }
}